lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -485,8 +485,19 @@ sub mark_job_linked {
     return undef if !$job || ($referer->path_query =~ /^\/?$/);
     my $comments = $job->comments;
     return undef if $comments->search({text => {like => 'label:linked%'}}, {rows => 1})->first;
+    return undef unless _specific_issue_exists($referer);
     my $user = $self->result_source->schema->resultset('Users')->system({select => ['id']});
     $comments->create_with_event({text => "label:linked Job mentioned in $referer_url", user_id => $user->id});
 }
 
+sub _specific_issue_exists {
+    my ($referer_url) = @_;
+    my $url = Mojo::URL->new($referer_url);
+    my $path = $url->path();
+    my ($issue_id) = $path =~ m{\d+};
```

Comment:
> Checking if the url path contains a number is a neat idea. Now, this wouldn't work with bugzilla urls `https://bugzilla.suse.com/show_bug.cgi?id=123` because the number is in the query string. We know that bugzilla somehow doesn't pass the full referrer anymore anyway, but this should ideally work for all kinds of urls, even if the issue id is in the query string.
> 
> What do others think?

I think it's a bad design to have such strict requirements for issue references in this code point. Please look into https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Utils.pm#L41

Reply:
Yes, I think I would also go for a mapping like the one for the bugrefs. That would of course change the way it currently works, so it should be done in a way it doesn't break existing installations.

Reply:
the link @okurz gave.
but i think we wouldn't even need a mapping, just a list.
maybe we can support both host names like before plus regexes that would match the whole url?
maybe if the string starts and ends with a `/` it is supposed to be uses as a regex? something like that...

Reply:
Still not sure what is the advantage of that mapping (or even list). We could achieve the same enhancing the regex (as it just checking for ID now, and as you said might not be sufficient for all, but still). it is simplier and you dont have to maintain another structure of data. 
Should I provide an alternative regex or feel free to implement another solution.

Reply:
This is what we have in our default config file:
```
#recognized_referers = bugzilla.suse.com bugzilla.opensuse.org progress.opensuse.org github.com
```
That's the list I mean. We can still use this, but I think we have to support regexes there.

Reply:
i take it as `provide an alternative regex`

Reply:
Rather "Make it possible to provide a regex in the configuration"

Reply:
Maybe we can simply use the `%BUGURLS` mapping. But that's hardcoded.
What do others think?

Reply:
i just saw the last two messages. 
the only thing I did is in https://github.com/os-autoinst/openQA/commit/71faa24c23161335b8c7a4a4a2d17305a406c099 but dont expect anything fancy.

Anyway I am waiting the existinting discussion to close to apply the other improvements. 

> Rather "Make it possible to provide a regex in the configuration"

if we go with this, I would also wish another PR for that

Reply:
Why a different PR? What would this PR here then be about?
Simply checking for digits is not solving any problem in general, it only fixes it specifically for `progress.opensuse.org`. But people can configure multiple referer hosts with different issue trackers, and for some your digit check would even break it.

I don't have a complete solution myself, but you need to think about the existing `recognized_referers` and should not develop something independent from that.

Reply:
> Maybe we can simply use the `%BUGURLS` mapping. But that's hardcoded. What do others think?

I can live with having the regexes in hardcoded in `%BUGURLS` and at the same time having an option to provide a custom "universal" regex in the configuration to cover a generic issue tracker we didn't cover.

Of course, the ideal option would be to put everything in the configuration. Maybe this is also big enough for its own section?

Instead of:
```ini
recognized_referers = bugzilla.suse.com progress.opensuse.org
```

something like:
```ini
[recognized_referers]
bugzilla.suse.com = id=(\d+)
progress.opensuse.org = issues/(\d+)
jira.example.com = (\w+-\d+)
```

Or is that too much?

Reply:
The other solution I was thinking of was
```
recognized_referers = /bugzilla.suse.com/show_bug.cgi?id=\d+/ some.other.host /https://progress.opensuse.org/issues/\d+/
```

But putting our bugref mapping into the configuration could be even better, and we don't actually need regexes because we already have everything we need:
```
bmo => 'https://bugzilla.mozilla.org/show_bug.cgi?id='
```

Reply:
On progress, it is `https://progress.opensuse.org/issues/new` vs `https://progress.opensuse.org/issues/123456` and I guess it could be similar for other issue tracker, so you definitely need more than a static prefix.

Reply:
After discussing with @baierjan I created a ticket to reimplement bugrefs via configuration: https://progress.opensuse.org/issues/178771
@AdamWill this might also be interesting for you and others, as you added some bugtracker urls in the past

Reply:
oh, yeah, that'd be great. i ran across another one that isn't implemented just the other day...qt bug tracker, I think.